lib/broadway/processor.ex

```diff
@@ -117,16 +117,34 @@ defmodule Broadway.Processor do
       module: module,
       context: context,
       processor_key: processor_key,
-      batchers: batchers
+      batchers: batchers,
+      name: name
     } = state
 
+    start_time = System.monotonic_time()
+
+    telemetry_metadata = %{
+      processor_key: processor_key,
+      name: name,
+      message: message
+    }
+
+    emit_message_start_event(start_time, telemetry_metadata)
+
     try do
-      module.handle_message(processor_key, message, context)
-      |> validate_message(batchers)
+      handle_message_result =
+        module.handle_message(processor_key, message, context)
+        |> validate_message(batchers)
+
+      emit_message_stop_event(start_time, telemetry_metadata)
+
+      handle_message_result
     catch
       kind, reason ->
         reason = Exception.normalize(kind, reason, __STACKTRACE__)
 
+        emit_message_error_event(start_time, telemetry_metadata, reason)
```

Comment:
The metadata for the error event should be: `kind: kind, reason: reason, stacktrace: __STACKTRACE__`. :)